### DISCUSSION TITLE
Password issue after localization settings

### DISCUSSION THREAD

**Discussion by tthenrie** [MichaIng/DietPi#6276] [Q&A]
Radxa4SE Radxa dietpi v8.15

As the default localization for dietpi is GB and the international 104key (GB) keyboard, and as the request for setting passwords comes early in the first-boot process, once the localization settings are re-set at the end, using the setup menus, it is possible that the user's passwords may not work after reboot. I ran into this problem, as my locale is US and my password for root and dietpi included symbols that use different keys in the International 104key (GB) and standard 104key (US) keyboards. After completing the first-boot and setup process I could not login. It took me awhile to figure out what the problem was.

I would suggest you _not_ request password changes during the first-boot process and leave that for the setup menu at the end, to obviate that potential problem.

**Comment by MichaIng** [comment]
You are right that this needs enhancement. The password change needs to stay as one of the first changes, but instead the keyboard layout change needs to be offered before one of the passwords is changed. This is however already tracked here: #5925
Good to have another case, so I'll stop moving this to next milestones but implement it for v8.16. Marking the duplicate however as closed.
